name-registry/SPECIFICATION.md

```diff
@@ -0,0 +1,66 @@
+Table of Contents
+- [Overview](#overview)
+- [Use Cases](#use-cases)
+  - [Actions that users are able to perform](#actions-that-users-are-able-to-perform)
+  - [Information that is presented to users](#information-that-is-presented-to-users)
+
+# Overview
+
+This document provides an overview of the application.
+
+It outlines the use cases, i.e. desirable functionality, in addition to requirements for the smart contract and the user interface.
+
+# Use Cases
+
+This section contains general information about the functionality of the application and thus does not touch upon any technical aspects.
+
+If you are interested in a functional overview then this is the section for you.
+
+## Actions that users are able to perform
+
+This sub-section details what a user is able to do e.g. click a button and "x, y, z" happens.
+
+### `register()`
+
+A user can register a name for themselves
+
+1. If the name is not already registered
+2. If the payment is sufficient for the duration
+3. If the payment is in the correct asset
+
+### `extend()`
+
+Any user can extend the registration duration of a given name
```

Comment:
Should mention that the name must be registered because otherwise it will revert